bokeh/embed/bundle.py

```diff
@@ -420,7 +420,33 @@ def _use_mathjax(objs: Sequence[Model | Document]) -> bool:
         bool
     '''
     from ..models.text import MathText
-    return _any(objs, lambda obj: isinstance(obj, MathText))
+
+    def model_requires_mathjax(model: Model) -> bool:
+        from ..models.axes import Axis
+
+        def is_tex_string(text: str) -> bool:
+            if text.startswith("$$") and text.endswith("$$"):
+                return True
+
+            elif text.startswith("\\[") and text.endswith("\\]"):
+                return True
+
+            elif text.startswith("\\(") and text.endswith("\\)"):
+                return True
+
+            return False
```

Comment:
Why does this need to be a closure? It seems like it could be a normal private helper function